### ISSUE TITLE
image orientation squeezes in shiny display

### ISSUE THREAD

**Issue by dschneiderch** [Gregor-Mendel-Institute/aradeepopsis#54]
It looks like the rosette carousel is expecting landscape images. when using portrait images it creates white space which makes everything smaller than it needs to be and hard to see.
![image](https://user-images.githubusercontent.com/7461221/103687576-586fed00-4f45-11eb-9a4c-336525aded12.png)

it would be nice to also have a choice how many images to view next to each other. on my labtop 3 I'd say I have room for 3 but on my external monitor i could probably comfortably view 6 or more.

in the rosette explorer it squeezes the images unnaturally
![image](https://user-images.githubusercontent.com/7461221/103687534-4a21d100-4f45-11eb-8638-1100c7e12484.png)

**Comment by phue** [comment]
Thanks for reporting this! I have to say that our images where always more or less quadratic, so I never noticed that.

> it would be nice to also have a choice how many images to view next to each other. on my labtop 3 I'd say I have room for 3 but on my external monitor i could probably comfortably view 6 or more.

Totally agree, I will look into it. Probably this can be made responsive to screen size.

**Comment by phue** [comment]
I made some changes to address this:

* https://github.com/Gregor-Mendel-Institute/aradeepopsis/commit/e8872c026c02c1c63a568ff8b676596636c1bd57 should fix the distorted images
* https://github.com/Gregor-Mendel-Institute/aradeepopsis/commit/f81c5613c0afc24f8bb2d3e83aa8eea1a0a1c56f adds breakpoints that change the number of images in the carousel according to screen width

It would be great if you could give this a try and let me know if it looks better now. The changes are on the `dev` branch, so to run it you would have to tell nextflow to use it like so:

`nextflow run Gregor-Mendel-Institute/aradeepopsis -r dev ...`

**Comment by dschneiderch** [comment]
i will try this on Friday

**Comment by dschneiderch** [comment]
I actually have time today but I was hoping to just run the new shiny app, if possible.
I ran the nextflow on the server and copied results to my computer to run the shiny.

I tried `docker run -v $(pwd):/mnt/shiny -p 44333:44333 beckerlab/aradeepopsis-shiny:dev R -e "shiny::runApp('/mnt/shiny/app.R', port=44333, host='0.0.0.0')"` so it downloads a different image labeled `dev` from dockerhub but I see the same issues.

**Comment by dschneiderch** [comment]
I copied the contents from [ dev branch app.R](https://github.com/Gregor-Mendel-Institute/aradeepopsis/blob/dev/assets/shiny/app.R) to my output directory and reran the docker.

I don't see a difference in the carousel
![image](https://user-images.githubusercontent.com/7461221/103928287-7ae24180-50d0-11eb-84d4-114c473eab2d.png)

The rosette explorer looks perfect
![image](https://user-images.githubusercontent.com/7461221/103928335-89305d80-50d0-11eb-9a2f-86c2de6a1baf.png)
although the color channels tab is quite stretched. I don' tknow if you can specify a max width, or you could increase the height. or leave it as good enough :-) It compresses when the window is resized.
![image](https://user-images.githubusercontent.com/7461221/103928500-c5fc5480-50d0-11eb-85df-2282b55ee7d9.png)

# rosette experiment
This was confusing because at first I got a new error:
I get a new error that looks like the data tried to join with itself.  I double checked my file and I only have 1 `class_norm_area` and it is of type double.
![image](https://user-images.githubusercontent.com/7461221/103928656-00fe8800-50d1-11eb-970c-9000ec43547a.png)

I noticed on line 29 of app.R you are reading in a fixed filename so I realized you are reading the original datafile and attempting to join the metadata. I actually only need to provide the metadata and not the traits themselves. I would suggest adding a note in the popup that no columns from the traits file should be in the metadata file except `file`

now it looks pretty good!
![image](https://user-images.githubusercontent.com/7461221/103929469-39eb2c80-50d2-11eb-8b9f-5bdc0617684b.png)

the one exception is that it doesn't recognize the factors as factors and instead orders them alphanumerically. I realize this is a constraint of saving to csv so i'm not sure how you'd improve that functionality.

**Comment by dschneiderch** [comment]
also, the yellow is pretty hard to see. I've found a background color of 'grey40' to work pretty well as a compromise for both dark and light colors.

**Comment by phue** [comment]
> I don't see a difference in the carousel

The number of images should now scale according to screen size. I can't do much about the whitespace between the images, that is how slickR renders the carousel thing and I can't seem to find an option to control this.

> the color channels tab is quite stretched

I changed the plot to be 600x600 and also increased the image size in the rosette explorer to be consistent 

> note in the popup that no columns from the traits file should be in the metadata file except

Good point, I added something like that. I guess it was not clear enough that the metadata is joined with the pipeline output.

> yellow is pretty hard to see

True, I changed the color map slightly and increased the line thickness


Thanks for your suggestions!
